common/persistence/sql/sqlTaskManager.go

```diff
@@ -71,15 +73,11 @@ func (m *sqlTaskManager) LeaseTaskQueue(request *persistence.LeaseTaskQueueReque
 	if err != nil {
 		return nil, serviceerror.NewInternal(err.Error())
 	}
-	shardID := m.shardID(nidBytes, request.TaskQueue)
-	namespaceID := request.NamespaceID
-	rows, err := m.db.SelectFromTaskQueues(&sqlplugin.TaskQueuesFilter{
-		ShardID:     shardID,
-		NamespaceID: &nidBytes,
-		Name:        &request.TaskQueue,
-		TaskType:    convert.Int64Ptr(int64(request.TaskType))})
+	tqId, tqHash := m.taskQueueIdAndHash(nidBytes, request.TaskQueue, request.TaskType)
```

Comment:
I see places like these where we take user request and compute TaskQueueId from it.  But I don't see any place where we deserialize TaskQueueId back into individual fields.  I'm assuming this is only used for querying database and all this information is already written as part of the info blob.  Just want to confirm this.

Reply:
Yea this is correct, we use the proto for all user returned values